cmd/find-main.go

```diff
@@ -0,0 +1,141 @@
+/*
+ * Minio Client (C) 2014, 2015 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"strings"
+
+	"github.com/fatih/color"
+	"github.com/minio/cli"
+	"github.com/minio/mc/pkg/console"
+)
+
+// find specific flags
+var (
+	findFlags = []cli.Flag{
+		cli.StringFlag{
+			Name:  "name",
+			Usage: "Find files of a matching name",
+		},
+		cli.StringFlag{
+			Name:  "path",
+			Usage: "Find global matches on the entire path",
+		},
+		cli.StringFlag{
+			Name:  "regex",
+			Usage: "Matches all files with PCRE regular expression",
+		},
+		cli.StringFlag{
+			Name:  "print",
+			Usage: "Prints the found files given args to STDOUT",
+		},
+		cli.StringFlag{
+			Name:  "mcexec",
```

Comment:
> What if we had two separate exec flags and the default for mcexec was {alias}, and the default for regular exec was {full}?

This might make things easier for writing code, but there is a lot of education on keeping it separate with all these variations - the price of that is immense. It might make the command itself unusable since we cannot easily convey what these mean. 

My suggestion is something more simple here keep it always `-exec`  and honor these styles. 

```
-exec "mc cp {} /mnt/pictures"
-exec "cp {} /mnt/pictures"
```

If user provides just `cp` then with auto substitution we treat it as `mc cp` , We never allow unix tools to be used in the beginning but can be used as part of the subsequent commands for example. 

```
-exec "mc share download --json {} | jq -r .share | xargs curl | convert - -resize "10%" - | mc pipe {}-thumb"
-exec "share download --json {} | jq -r .share | xargs curl | convert - -resize "10%" {}-thumb.png"
```

Here `share` just becomes `mc share`